sdk/clientcore/annotation-processor-test/src/main/java/io/clientcore/annotation/processor/test/SpecialReturnBodiesServiceImpl.java

```diff
@@ -147,4 +151,31 @@ public Response<InputStream> getInputStreamWithResponse(String url) {
         }
         return new Response<>(networkResponse.getRequest(), responseCode, networkResponse.getHeaders(), networkResponse.getValue().toStream());
     }
+
+    @SuppressWarnings("cast")
+    @Override
+    public Response<List<BinaryData>> getListOfBinaryData(String endpoint) {
+        // Create the HttpRequest.
+        HttpRequest httpRequest = new HttpRequest().setMethod(HttpMethod.GET).setUri(endpoint + "/type/array/unknown");
+        // Send the request through the httpPipeline
+        Response<BinaryData> networkResponse = this.httpPipeline.send(httpRequest);
+        int responseCode = networkResponse.getStatusCode();
+        boolean expectedResponse = responseCode == 200;
+        if (!expectedResponse) {
+            String errorMessage = networkResponse.getValue().toString();
+            networkResponse.close();
+            throw new HttpResponseException(errorMessage, networkResponse, null);
+        }
+        List<BinaryData> deserializedResult;
+        ParameterizedType returnType = CoreUtils.createParameterizedType(List.class, BinaryData.class);
+        SerializationFormat serializationFormat = CoreUtils.serializationFormatFromContentType(httpRequest.getHeaders());
+        if (jsonSerializer.supportsFormat(serializationFormat)) {
+            deserializedResult = CoreUtils.decodeNetworkResponse(networkResponse.getValue(), jsonSerializer, returnType);
+        } else if (xmlSerializer.supportsFormat(serializationFormat)) {
+            deserializedResult = CoreUtils.decodeNetworkResponse(networkResponse.getValue(), xmlSerializer, returnType);
```

Comment:
This won't work as `JsonSerializer` and `XmlSerializer` don't know how to handle `BinaryData` as it isn't declared as a serializable type. We'll likely need to figure out a comprehensive handling for this that isn't hardcoding support for `BinaryData` into those serializers.

Reply:
Added a follow up issue to add this support https://github.com/Azure/azure-sdk-for-java/issues/45354